pkg/virt-api/webhooks/validating-webhook/admitters/vmrestore-admitter.go

```diff
@@ -199,30 +198,6 @@ func (admitter *VMRestoreAdmitter) validateCreateVM(field *k8sfield.Path, vmRest
 		return nil, nil, false, err
 	}
 
-	rs, err := vm.RunStrategy()
```

Comment:
Why is removing the webhook make it non imperative. Once VM is stopped we have the finalizer preventing from the VM to change during the action.. But why should we prevent the creation the the VMRestore object if we can create it and present in status that the process cant continue before the vm is stopped? thats against k8s eventual consistency model..
In order to be consistent with any other webhook I dont see webhook warning about this kind of stuff, its always being updated in the conditions.. I am not against replacing the rejection with a log of warning I just doing think its the norm in k8s and kubevirt.

Reply:
Given the way things used to work (reject if vm running) and acknowledging that users may be relying on that behavior (this API has been around for 4+ years) I think it would be nice to at least give users a heads up that the implementation has completely changed especially now that we are beta

Perhaps you can bring this up at the next kubevirt sig-api meeting?  I am truthfully not fully convinced that in this case it is more important to satisfy eventual consistency purists at the detriment to users of a long standing API

Reply:
Although I always try to be careful about changing longstanding API behavior (see previous reply), the thing that actually concerns me the most with this PR is the fact that a restore may take an indeterminate time to "activate."  Can we do something else?  These are all things that I think are better and should satisfy the eventual consistency heads:

1.  Restore controller immediately marks the restore Failed if the VM is running
2.  Similar to snapshot deadline, the restore can be configured to be time bounded and wait for graceful shutdown before going to Failed
3.  Terminate the running VM and restore immediately
4.  Like 2 but instead of failing, terminate VM and restore 

If I could only choose one, I would pick 3 but it may be nice to have options